### DISCUSSION TITLE
Warning / Linting about routeLoader left behind

### DISCUSSION THREAD

**Discussion by czuma** [QwikDev/qwik-evolution#17] [External Library Ideas]
#### Problems you are trying to solve:
* If you create a routeLoader, this will be called by qwik-city, even if the child components not even make use of them.
* if you make a big refactor of components to stop using the routeLoader and forget remove the routeLoader from your layout files, they will still use resources.
* 

#### Goals you are trying to achieve:
* prevent unwanted code excecution

#### Any other context or information you want to share:

---

#### What do you propose?
on dev/build/preview generate an error or a warning if you have routeLoaders not being used in any part of your code

#### Code examples
put this inside a random index.tsx or layout and have fun trying to locate them
you don't even need to call it anywhere.

``` 
export const useDandForget = routeLoader$(async () => {
 console.log('you...sneaky...bug y found you, you eat my cpu credits!')
});
```

---

**Comment by wmertens** [comment]
Good idea but I wonder how to find uses. Perhaps after the build just scan for the qrl symbol and check that it's being called (ignoring where it's defined)

**Comment by shairez** [comment]
Thanks for the proposal @czuma !

We've reviewed this proposal and came to the following conclusion:

Figuring out if a loader is actually being used can be a very hard problem because it cannot be guaranteed only by static analysis.
So adding this warning as an error / warning in the compilation is impossible. 

But, maybe writing a lint rule that simply checks if a specific symbol (`useDangForget`) is not being used anywhere and can warn about it might be valuable. 

We don't have the resources to create / maintain it, so we thought it would be valuable to implement as a community plugin for the linter. 

If you or anyone else wants to take this on, this could be very helpful.

So I'm moving it to "External Library Ideas".

Thanks again! 🙏

**Comment by wmertens** [comment]
This could be part of a report on the API that should get autogenerated and
committed.

It could list all server$, action$ calls with their scope and parameters
and all middleware/onXyz/route loaders

Then when refactoring to remove a route loader you should expect that file
to change.
Wout.
